### PR TITLE
Fix logout at profile page

### PR DESCRIPTION
**Pull request by NguyenAnhQuoc1994**

### CODE REVIEW

File: src/Pages/Profile/Profile.tsx [hunk 1]
```diff
@@ -1,18 +1,38 @@
...
+      userId: null,
+    }
   }
 
-  public render() {
+  public componentWillMount() {
+    this.checkCurrentUser();
+  }
+
+  public checkCurrentUser() {
```
Comment (lybaokhanh) [inline src/Pages/Profile/Profile.tsx]: Please using another name to make it clear.

File: src/Pages/Profile/Profile.tsx [hunk 2]
```diff
@@ -1,18 +1,38 @@
...
     const userId = this.props.match.params.userId;
+    const userInfo = localStorageManager.getUserInfo();
+    if (!userInfo || userInfo.id !== userId) {
+      this.setState({userId});
+    }
+  }
+
+  public render() {
+    const userId = this.state.userId;
```
Comment (lybaokhanh) [inline src/Pages/Profile/Profile.tsx]: We should use destructuring assignment `const { userId } = this.state;` to unpack value.

File: src/Modules/User/Components/Profile/UserDropdown/UserDropdown.tsx [hunk 3]
```diff
@@ -189,6 +204,7 @@ const mapDispatchToProps = (dispatch: Dispatch) => ({
     dispatch(updateUserInfo(userInfo)),
 });
 
-export const UserDropdown = connect(mapStateToProps, mapDispatchToProps)(
-  UserDropdownComponent,
-);
+export const UserDropdown = withRouter(connect(
```
Comment (lybaokhanh) [inline src/Modules/User/Components/Profile/UserDropdown/UserDropdown.tsx]: Please wrap `connect` and `withRouter` into `compose` utility

**Review by lybaokhanh — CHANGES_REQUESTED** [review]

File: src/Pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -1,18 +1,41 @@
...
 
-  public render() {
+  public componentWillMount() {
     const userId = this.props.match.params.userId;
+    if (!this.isCurrentUser(userId)) {
+      this.setState({ userId });
+    }
+  }
+
+  public isCurrentUser(userId: string) {
```
Comment (lybaokhanh) [inline src/Pages/Profile/Profile.tsx:29]: set `private` function if it will not be reused

**Comment by NguyenTatNhac** [comment]
I think you should close this PR, since we already have a task "reload whenever login or logout", so we will combine that 2 PRs. 
Thanks!

**Comment by NguyenAnhQuoc1994** [comment]
I think so. I close it and will create a new PR after finishing task "reload whenever login or logout".
Thanks you guys!

**Pull request closed without merging**